src/shared/acl-util.c

```diff
@@ -209,26 +209,50 @@ int acl_search_groups(const char *path, char ***ret_groups) {
         return ret;
 }
 
-int parse_acl(const char *text, acl_t *ret_acl_access, acl_t *ret_acl_default, bool want_mask) {
-        _cleanup_free_ char **a = NULL, **d = NULL; /* strings are not freed */
+int parse_acl(
+                const char *text,
+                acl_t *ret_acl_access,
+                acl_t *ret_acl_access_cond_exec,
+                acl_t *ret_acl_default,
+                bool want_mask) {
+
+        _cleanup_free_ char **a = NULL, **e = NULL, **d = NULL; /* strings are not freed */
         _cleanup_strv_free_ char **split = NULL;
         int r = -EINVAL;
-        _cleanup_(acl_freep) acl_t a_acl = NULL, d_acl = NULL;
+        _cleanup_(acl_freep) acl_t a_acl = NULL, e_acl = NULL, d_acl = NULL;
 
         assert(text);
         assert(ret_acl_access);
+        assert(ret_acl_access_cond_exec);
         assert(ret_acl_default);
 
         split = strv_split(text, ",");
         if (!split)
                 return -ENOMEM;
 
         STRV_FOREACH(entry, split) {
+                _cleanup_strv_free_ char **entry_split = NULL;
+                _cleanup_free_ char *entry_join = NULL;
                 char *p;
+                int n;
 
-                p = STARTSWITH_SET(*entry, "default:", "d:");
+                n = strv_split_full(&entry_split, *entry, ":", EXTRACT_RETAIN_SEPARATORS|EXTRACT_DONT_COALESCE_SEPARATORS|EXTRACT_RETAIN_ESCAPE);
+                if (n < 0)
+                        return -ENOMEM;
+
+                if (n < 3)
+                        return -EINVAL;
+
+                string_replace_char(entry_split[n-1], 'X', 'x');
+                entry_join = strv_join(entry_split, ":");
+                if (!entry_join)
+                        return -ENOMEM;
+
+                p = STARTSWITH_SET(entry_join, "default:", "d:");
                 if (p)
                         r = strv_push(&d, p);
+                else if (!streq(*entry, entry_join))
+                        r = strv_push(&e, entry_join);
```

Comment:
maybe use `r = strv_push(&e, TAKE_PTR(entry_join))` to donate the allocated string here.